addons/purchase_mrp/tests/test_purchase_mrp_flow.py

```diff
@@ -1251,3 +1251,70 @@ def test_kit_price_without_rounding(self):
         # however, due to rounding differences, the expected value is 100
         svl_val = self.env['stock.valuation.layer'].search([('stock_move_id', '=', move.id)]).value
         self.assertEqual(svl_val, 100)
+
+    def test_purchase_kit_bill_before_reception_component_cost_exactly_aligns_with_kit_product_cost(self):
+        """ When a kit product is invoiced prior to delivery, we want to make sure to reconcile all
+        the AMLs from its explosion together, else we risk re-reconciliation attempts (which will
+        block certain actions from being performed altogether).
+        """
+        kit_product = self.env['product.product'].create({
+            'name': 'kit prod',
+            'purchase_method': 'purchase',
+            'type': 'product',
+            'standard_price': 10,
+            'list_price': 20,
+        })
+        kit_product.categ_id.write({
+            'property_cost_method': 'average',
+            'property_valuation': 'real_time',
+        })
+        components = self.env['product.product'].create([{
+            'name': f'comp {i}',
+            'type': 'product',
+            'standard_price': 5,
+            'list_price': 5,
+        } for i in (1, 2)])
+        self.env['mrp.bom'].create({
+            'type': 'phantom',
+            'product_id': kit_product.id,
+            'product_tmpl_id': kit_product.product_tmpl_id.id,
+            'product_qty': 1,
+            'bom_line_ids': [Command.create({
+                'product_id': comp.id,
+                'product_qty': 1,
+            }) for comp in components
+        ]})
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': self.partner_a.id,
+            'order_line': [Command.create({
+                'product_id': kit_product.id,
+                'product_qty': 1,
+            })],
+        })
+        purchase_order.button_confirm()
+        purchase_order.action_create_invoice()
+        bill = purchase_order.invoice_ids
+        bill.invoice_date = fields.Date.today()
+        bill.action_post()
+        receipt = purchase_order.picking_ids
+        # would fail due to attempted re-reconciliation prior to this commit
+        receipt.button_validate()
+        stock_input_account, stock_valuation_account, tax_paid_account, account_payable_account = (
+            kit_product.categ_id.property_stock_account_input_categ_id,
+            kit_product.categ_id.property_stock_valuation_account_id,
+            self.company_data['default_account_tax_purchase'],
+            self.company_data['default_account_payable'],
+        )
+        # stock input account move lines should be reconciled
+        self.assertRecordValues(
+            self.env['account.move.line'].search([], order='id asc'),
```

Comment:
ok for the fix. Could I ask you to add product_id in the aml values to assert ? thank you